vtex/node-vtex-api
# Change tracing export. Add tracing config to all clients requests

#### What is the purpose of this pull request?
- Add `NODE_ENV` tag to process tracing tags.
- Add tracingConfig to all clients requests.
- Create tracing config on requests to specify if the request span will be child of the root span or will follow from (Refer to `References between Spans` on https://opentracing.io/specification/)
- Create tracing config on requests to specify a suffix to the request span name - it will be `http-request:suffix`
- Change way to export tracing utilities - this way seemed a little off (it is a breaking change, but just vtex.monitoring-proxy is supposed to be using this) 
- Create helper `createTracingContextFromCarrier` to allow worker threads and child_processes to  be traced.

#### How should this be manually tested?
Deploy this version on a test cluster and check for example the spans on `vtex.render-server`

#### Types of changes
